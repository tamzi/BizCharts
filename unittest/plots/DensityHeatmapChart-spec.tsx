import React, { useEffect, useState } from 'react';
import DensityHeatmapChart from '../../src/plots/DensityHeatmapChart';
import { render, cleanup, act } from '@testing-library/react';

function Demo(props) {
  const [data, setData] = useState([]);
  useEffect(() => {
    fetch('https://alifd.alibabausercontent.com/materials/@bizcharts/heatmap-image/0.3.0/mock.json')
      .then(res => res.json())
      .then(data => {
        act(() => setData(data));
      })
  }, [])
  
  return (
    <DensityHeatmapChart
      data={data}
      title={{
        visible: true,
        text: '密度热力图',
      }}
      autoFit
      {...props}
    />
  );
}

describe('Plots-DensityHeatmapChart', () => {
  test('旧版密度热力图', () => {
    let chart = null;
    render(<Demo
      padding="auto"
      xField='g'
      yField='l'
      colorField='tmp'
      color={['#295599', '#3e94c0', '#78c6d0', '#b4d9e4', '#fffef0', '#f9cdac', '#ec7d92', '#bc448c']}
      radius={15}
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
  test('基础密度热力图', () => {
    let chart = null;
    render(<Demo
      padding="auto"
      xField='g'
      yField='l'
      colorField='tmp'
      color="#F51D27-#FA541C-#FF8C12-#FFC838-#FAFFA8-#80FF73-#12CCCC-#1890FF-#6E32C2"
      radius={15}
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  });
})
