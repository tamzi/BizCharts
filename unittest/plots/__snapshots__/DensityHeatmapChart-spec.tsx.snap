// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-DensityHeatmapChart 基础密度热力图 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": "#F51D27-#FA541C-#FF8C12-#FFC838-#FAFFA8-#80FF73-#12CCCC-#1890FF-#6E32C2",
  "colorField": "tmp",
  "data": Array [],
  "legend": false,
  "padding": "auto",
  "radius": 15,
  "renderer": "canvas",
  "sizeRatio": 15,
  "type": "density",
  "xAxis": Object {
    "grid": Object {
      "alignTick": false,
      "line": Object {
        "style": Object {
          "lineDash": null,
          "lineWidth": 1,
          "stroke": "#f0f0f0",
        },
      },
    },
    "label": Object {
      "autoHide": Object {
        "cfg": Object {
          "minGap": 6,
        },
        "type": "equidistance",
      },
      "autoRotate": false,
    },
    "line": null,
    "nice": true,
    "tickLine": null,
  },
  "xField": "g",
  "yAxis": Object {
    "grid": Object {
      "alignTick": false,
      "line": Object {
        "style": Object {
          "lineDash": null,
          "lineWidth": 1,
          "stroke": "#f0f0f0",
        },
      },
    },
    "label": Object {
      "autoHide": true,
      "autoRotate": false,
    },
    "nice": true,
  },
  "yField": "l",
}
`;

exports[`Plots-DensityHeatmapChart 旧版密度热力图 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": Array [
    "#295599",
    "#3e94c0",
    "#78c6d0",
    "#b4d9e4",
    "#fffef0",
    "#f9cdac",
    "#ec7d92",
    "#bc448c",
  ],
  "colorField": "tmp",
  "data": Array [],
  "legend": false,
  "padding": "auto",
  "radius": 15,
  "renderer": "canvas",
  "sizeRatio": 15,
  "type": "density",
  "xAxis": Object {
    "grid": Object {
      "alignTick": false,
      "line": Object {
        "style": Object {
          "lineDash": null,
          "lineWidth": 1,
          "stroke": "#f0f0f0",
        },
      },
    },
    "label": Object {
      "autoHide": Object {
        "cfg": Object {
          "minGap": 6,
        },
        "type": "equidistance",
      },
      "autoRotate": false,
    },
    "line": null,
    "nice": true,
    "tickLine": null,
  },
  "xField": "g",
  "yAxis": Object {
    "grid": Object {
      "alignTick": false,
      "line": Object {
        "style": Object {
          "lineDash": null,
          "lineWidth": 1,
          "stroke": "#f0f0f0",
        },
      },
    },
    "label": Object {
      "autoHide": true,
      "autoRotate": false,
    },
    "nice": true,
  },
  "yField": "l",
}
`;
